docs/pages/versions/unversioned/sdk/app-integrity.mdx

```diff
@@ -0,0 +1,186 @@
+
+> **Note** To use the App Attest service, your app must have an App ID that you register on the Apple Developer website.
+
+For verification logic on your server, see [Validating apps that connect to your server](https://developer.apple.com/documentation/devicecheck/validating-apps-that-connect-to-your-server).
+
+### Check if the device supports App Attestation
+
+Not all devices can use the App Attest service, so it’s important to have your app run a compatibility check before accessing the service. If the user’s app doesn’t pass the compatibility check, gracefully bypass the service. You check for availability by reading the `isSupported` property.
+
+```js
+import * as AppIntegrity from '@expo/app-integrity';
+
+if (AppIntegrity.isSupported) {
+  // Perform key generation and attestation.
+}
+// Continue with your server API access.
+```
+
+> **Note** App Attest is not supported on simulator.
+
+> **important**  Most app extensions don’t support App Attest. Generally, when executing code in these extensions, bypass key generation and attestation, even if the `isSupported` method property is `true`. The only app extensions that support App Attest are watchOS extensions in watchOS 9 or later. For these extensions, you can use the results from `isSupported` to indicate whether your WatchKit extension bypasses attestation.
+
+
+### Create a key pair
+
+For each user account on each device running your app, generate a unique, hardware-based, cryptographic key pair by calling the `generateKey` method.
+
+```js
+const keyId = await AppIntegrity.generateKey()
+```
+
+On success, the method returns a key identifier (`keyId`) that you use later to access the key. Record the identifier in persistent storage because there’s no way to use the key without the identifier, and no way to get the identifier later. The device automatically stores the associated private key in the Secure Enclave, from where the App Attest service can use it to create signatures, but from where no process can ever directly read or modify it, ensuring its security.
+
+> **Important** If you create a key pair in an App Clip, use the same key pair in the corresponding app. To support this, be sure to store the identifier in a shared container accessible from your full app.
+
+Don’t reuse a key among multiple users on a device because this weakens security protections. In particular, it becomes hard to detect an attack that uses a single compromised device to serve multiple remote users running a compromised version of your app. For more information, see [Assessing fraud risk](https://developer.apple.com/documentation/devicecheck/assessing-fraud-risk).
+
+### Get a challenge from your server
+
+Request a unique, one-time challenge from your server. This challenge will be embedded in the attestation step below, ensuring it can’t be reused by an attacker. The challenge should be at least 16 bytes in length to ensure sufficient entropy to ensure guessing them is infeasible.
```

Comment:
```suggestion
Request a unique, one-time challenge from your server. This challenge will be embedded in the attestation step below, ensuring it can't be reused by an attacker. The challenge should be at least 16 bytes long to provide enough entropy so that guessing it is infeasible.
```

Let's improve the readability of the last sentence.